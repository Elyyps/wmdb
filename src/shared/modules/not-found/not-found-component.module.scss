.not-found {
  width: 100%;
  height: calc(100vh - 200px);
  @media (max-width: $breakpoint-xsmall-max) {
    height: calc(100vh - 370px);
    min-height: calc(100vh - 370px);
    max-height: calc(100vh - 370px);
    
  }
  text-align: center;
  color: white !important;
  display: table;
  align-items: center;
  &__table-cell {
    display: table-cell;
    align-items: center;
    vertical-align: middle;
  }
  &__container {
    display: flex;
    flex-direction: row;
    margin: auto;
    width: 40%;
    @media (max-width: $breakpoint-medium-max) {
      width: 60%;
    }
    @media (max-width: $breakpoint-xsmall-max) {
      width: 80%;
      flex-direction: column;
      text-align: center;
      flex-direction: column-reverse;
      > span {
        margin: auto;
        margin: 16px auto;
        height: 190px !important;
      }
    }
  }
  &__content {
    text-align: left;
    margin-top: 10px;
    margin-left: 36px;
    &-title {
      h2 {
        color: white !important;
        @media (max-width: $breakpoint-xsmall-max) {
          margin-top: 32px;
        }
      }
      margin-bottom: 0;
    }
    @media (max-width: $breakpoint-xsmall-max) {
      text-align: center;
      button {
        width: 100% !important;
        margin-top: 16px !important;
      }
      margin: 0;
    }
  }
  button {
    display: block;
    margin-top: 41px;
  }
  background-image: linear-gradient(to right, #34aadf, #02906f);
  svg {
    path {
      fill: white !important;
      stroke: white !important;
    }
  }
}
