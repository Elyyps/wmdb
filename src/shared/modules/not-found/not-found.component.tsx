import * as React from "react";
import styles from "./not-found-component.module.scss";
import PlaceholderImage from "@assets/WMDB2.svg";
import { IconComponent } from "@app/core/icon";
import { Button } from "@app/core/button";

export interface INotFoundComponentProps {}

const NotFoundComponent = (props: INotFoundComponentProps) => (
  <div className={styles["not-found"]}>
    <div className={styles["not-found__table-cell"]}>
      <div className={styles["not-found__container"]}>
        <IconComponent height={"220px"} width="160px" fillColor="white" icon={PlaceholderImage} />
        <div className={styles["not-found__content"]}>
          <div className={styles["not-found__content-title"]}>
            <h2>Oeps!</h2>
          </div>
          <span>Pagina niet gevonden. We helpen je graag de weg weer terug te vinden.</span>
          <Button title="WMDB homepage" />
        </div>
      </div>
    </div>
  </div>
);

export { NotFoundComponent };
