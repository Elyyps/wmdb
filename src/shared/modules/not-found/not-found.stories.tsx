import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { NotFoundComponent } from "./not-found.component";

storiesOf("NotFound", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <NotFoundComponent />);
