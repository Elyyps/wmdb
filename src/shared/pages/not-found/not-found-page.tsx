import * as React from "react";
import { NavBarComponent } from "@app/modules/nav-bar";
import { generateNavbarData } from "@app/api/modules/nav-bar/end-point";
import { FooterComponent } from "@app/modules/footer";
import { generateFooterData } from "@app/api/modules/footer/end-point";
import { NotFoundComponent } from "@app/modules/not-found";

const NotFoundPageComponent = () => (
  <React.Fragment>
    <NavBarComponent navbarModule={generateNavbarData()} />
    <NotFoundComponent />
    <FooterComponent footerModule={generateFooterData()} />
  </React.Fragment>
);
export { NotFoundPageComponent };
