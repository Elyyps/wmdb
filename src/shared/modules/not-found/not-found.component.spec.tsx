import React from "react";
import { shallow } from "enzyme";
import { NotFoundComponent } from "./not-found.component";

describe("[NotFound]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<NotFoundComponent />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
