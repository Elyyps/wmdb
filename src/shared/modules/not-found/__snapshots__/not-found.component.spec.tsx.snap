// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[NotFound] should render component without crashing 1`] = `
<div
  className="not-found"
>
  <div
    className="not-found__table-cell"
  >
    <div
      className="not-found__container"
    >
      <IconComponent
        fillColor="white"
        height="220px"
        icon="WMDB2.svg"
        width="160px"
      />
      <div
        className="not-found__content"
      >
        <div
          className="not-found__content-title"
        >
          <h2>
            Oeps!
          </h2>
        </div>
        <span>
          Pagina niet gevonden. We helpen je graag de weg weer terug te vinden.
        </span>
        <Button
          title="WMDB homepage"
        />
      </div>
    </div>
  </div>
</div>
`;
